### ISSUE TITLE
pub global activate via path not working unless `.dart_tool/pub/bin/<executable>` is deleted

### ISSUE THREAD

**Issue by btrautmann** [dart-lang/sdk#56777]
```
If providing this information as part of reporting a bug, please review the information
below to ensure it only contains things you're comfortable posting publicly.

#### General info

- Dart 3.5.0 (stable) (Tue Jul 30 02:17:59 2024 -0700) on "macos_arm64"
- on macos / Version 14.7 (Build 23H124)
- locale is en-US

#### Project info

- sdk constraint: '^3.5.0'
- dependencies: 
- dev_dependencies: lints, test

#### Process info

|  Memory |  CPU | Elapsed time | Command line                                                                    |
| ------: | ---: | -----------: | ------------------------------------------------------------------------------- |
|   23 MB | 0.0% |     22:57:26 | dart devtools --machine --allow-embedding --dtd-uri ws:<path>/SLOui8f56EyIjsj7  |
| 2278 MB | 0.0% |     22:57:27 | dart language-server --protocol=lsp --client-id=VS-Code --client-version=3.96.0 |
|  464 MB | 0.0% |        03:18 | dart language-server --protocol=lsp --client-id=VS-Code --client-version=3.96.0 |
|   22 MB | 0.0% |     22:57:27 | dart tooling-daemon --machine                                                   |
|   82 MB | 0.0% |        03:18 | dart tooling-daemon --machine                                                   |
|   49 MB | 0.3% |     22:57:27 | flutter_tools.snapshot daemon                                                   |
```

## The Issue

I started noticing a few weeks ago that `dart pub global activate --source="path" <path>` stopped working. To reproduce, you can:

1. use a project created via `dart create -t console <name>`
2. add the following to the `pubspec.yaml`
```yaml
executables:
  name: name
```
3. activate the executable globally via `dart pub global activate --source="path" <path/to/root/of/project>
4. Run the executable via `name` and note result
5. Change the logic of the executable (in the case of the `dart create -t console` default project, update `calculate()` to return a different integer.
6. Repeat step 3-4, noting that the result has not changed. 
7. Delete the `.dart_tool/pub/bin/<name>` directory
8. Repeat steps 3-4, noting that the result does change this time.

### Expected results

Prior to this bug, re-running `dart pub global activate --source="path" would correctly update the activated executable such that the new logic would be respected.

**Comment by btrautmann** [comment]
> **Summary:** The user reports that `dart pub global activate --source="path"` no longer updates activated executables when the executable's logic is changed. Deleting the `.dart_tool/pub/bin/<executable>` directory before activating the executable again resolves the issue.

Good bot

**Comment by btrautmann** [comment]
@bkonyi can you LMK if this fell through the cracks? This is a pretty annoying bug to run into, particularly in cases like self-hosted github actions runners where state is not wiped between runs and so implementation hangs around until you remember this is a thing. I imagine there are others scratching their heads as to why `pub global activate` is not working. 👀

**Comment by bkonyi** [comment]
Yeah, I'm not sure if the pub team is subscribed to the `dart-cli-pub` label, so I'll just cc @sigurdm and @jonasfj to make sure they're aware.
